docker/bookkeeper/entrypoint.sh

```diff
@@ -48,31 +56,143 @@ export BK_tlsTrustStoreType=JKS
-  # The container already contains data in BK directories. This is probably because
-  # the container has been restarted; or, if running on Kubernetes, it has probably been
-  # updated or evacuated without losing its persistent volumes.
-  echo "data available in bookkeeper directories; not formatting the bookie"
-else
-  # The container does not contain any BK data, it is probably a new
-  # bookie. We will format any pre-existent data and metadata before starting
-  # the bookie to avoid potential conflicts.
-  echo "format bookie data and metadata"
-  /opt/bookkeeper/bin/bookkeeper shell bookieformat -nonInteractive -force -deleteCookie
-fi
-
-echo "start bookie"
-/opt/bookkeeper/scripts/entrypoint.sh bookie
+# Create directories for multiple ledgers and journals if specified.
+create_bookie_dirs() {
+  IFS=',' read -ra directories <<< $1
+  for i in "${directories[@]}"
+  do
+      mkdir -p $i
+      if [ "$(id -u)" = '0' ]; then
+          chown -R "${BK_USER}:${BK_USER}" $i
+      fi
+  done
+}
+
+wait_for_zookeeper() {
+    echo "Waiting for zookeeper"
+    until zk-shell --run-once "ls /" ${BK_zkServers}; do sleep 5; done
+}
+
+
+create_zk_root() {
+    if [ "x${BK_CLUSTER_ROOT_PATH}" != "x" ]; then
+        echo "Creating the zk root dir '${BK_CLUSTER_ROOT_PATH}' at '${BK_zkServers}'"
+        zk-shell --run-once "create ${BK_CLUSTER_ROOT_PATH} '' false false true" ${BK_zkServers}
+    fi
+}
+
+configure_bk() {
+    # We need to update the metadata endpoint and Bookie ID before attempting to delete the cookie
+    sed -i "s|.*metadataServiceUri=.*\$|metadataServiceUri=${BK_metadataServiceUri}|" /opt/bookkeeper/conf/bk_server.conf
+    if [ ! -z "$BK_useHostNameAsBookieID" ]; then
+      sed -i "s|.*useHostNameAsBookieID=.*\$|useHostNameAsBookieID=${BK_useHostNameAsBookieID}|" ${BK_HOME}/conf/bk_server.conf
+    fi
+}
+
+fix_bk_ipv6_check() {
+
+   # `${SCRIPTS_DIR}/common.sh` - a shell script in Bookeeper, runs this command: `/sbin/sysctl -n net.ipv6.bindv6only`.
+   # This command can return an error exit code `255` and error message:
+   #    `sysctl: cannot stat /proc/sys/net/ipv6/bindv6only: No such file or directory`
+   #
+   # If the `common.sh` file is sourced in a script that sets "exit on error" as true (set -e), the script shall return
+   # an error code `255`. Since, the `bookkeeper/bin/bookeeper` shell script sources the `common.sh` file, and it sets
+   # "exit on error" as true, we encounter this error when it is invoked. To avoid that error, here we modify the
+   # `common.sh` file to execute `/sbin/sysctl -n net.ipv6.bindv6only` only if file `/proc/sys/net/ipv6/bindv6only`
+   # is available.
+
+   # Replace the 22nd line with "if [ -f /sbin/sysctl ] && [ -f /proc/sys/net/ipv6/bindv6only ]; then".
+   sed -i "22s|.*|if [ -f /sbin/sysctl ] \&\& [ -f /proc/sys/net/ipv6/bindv6only ]; then|" ${BK_HOME}/bin/common.sh
```

Comment:
The purpose of check `/sbin/sysctl -n net.ipv6.bindv6only` in common.sh, is to error-out if IPV6 is enabled - indicated by `sysctl -n net.ipv6.bindv6only` giving ouput `net.ipv6.bindv6only=1`.
In our case this command fails because the key "net.ipv6.bindv6only" is not set in sysctl. IMO, instead of checking for existence of file `/proc/sys/net/ipv6/bindv6only` which is an implementation detail of `sysctl` command, it may be better to skip failing the script if output of `/sbin/sysctl -n net.ipv6.bindv6only` is non zero, indicate command did not succeed.
This can be done by adding "set -e" @ ln 23 and "set +e" at line 25 in [common.sh](https://github.com/apache/bookkeeper/blob/release-4.9.2/bin/common.sh#L25), so that the script does not exit when `/sbin/sysctl -n net.ipv6.bindv6only` gives non-zero exit code.

Reply:
Turning off exit-on-error flag (`set +e`) in the `common.sh` temporarily before executing [this ](https://github.com/apache/bookkeeper/blob/release-4.9.2/bin/common.sh#L25) line in is only a workaround. So is true with what is being done in the code referenced above (where we are replacing the if condition to include check for `/proc/sys/net/ipv6/bindv6only`. 

IMO, the latter is better due to the following: 
* It modifies the specific line that has the problem and therefore the workaround is more explicit about what the problem is.
* It is closer to what the ultimate fix may be. Witness [this ](https://github.com/apache/bookkeeper/blob/a7855d994ee70b22d12223dc14da602da9b26196/bin/common.sh#L21) code in a Bookkeeper PR that I've raised.  Note that ultimately, the `set -e` is done in other Bookkeeper scripts that invoke the `common.sh` (and not in the `common.sh` file itself), so adding these flags in the `common.sh` file would be odd.

Reply:
I do not agree that adding "set -e" is a "workaround" and there is nothing "odd" about using it.
This the standard way to ignore failures in commands that are likely to fail for reasons unrelated to the script.
Infact checking for  "/proc/sys/net/ipv6/bindv6only" is a **hack** as the location of that file may be different on different platforms.

Reply:
It is workaround in the sense that we let the error occur and then ignore it. The change in condition, on the other hand, prevents the error from happening in the first place, and is still a workaround in the sense that it is compensating for the bug in `common.sh`. 

Are you sure that the path `/proc/sys/net/ipv6/bindv6only` is different on different platforms? And, can this path change with the Bookkeeper container running on those platforms, given that it is based of centos (https://github.com/apache/bookkeeper/blob/44a1a4e986e22d87ba340069a7b5619feb30f0dc/docker/Dockerfile#L20)?

Reply:
I'm in general skeptical of ignoring errors because it is easy to overlook some important error and get bitten at some point. If we can preserve exit-on-error while ignoring only the specific error we are trying to, then I'd say that's a safer option.

Reply:
@pbelgundi I've found that the procfs path is really a pseudo file-system that exists only on memory. See https://wiki.archlinux.org/index.php/Procfs. So, we can indeed depend on the presence on `/proc/sys/`. 

@fpj I agree with your last statement and believe that the current code is in line with it. I suppose you aren't proposing any change.

Reply:
@fpj we're not  **ignoring** an error here, but avoiding the script from exiting because a certain command fails in the absence of `IPV6`, which is a very valid thing to do....
I still think adding "set -e"  and "set +e" is the right way to solve the problem.

However, it isn't so severe that I would want to block the approval of this PR for the same.
@ravisharda I would suggest you think through this and decide.